Code/Framework/AzCore/AzCore/DOM/DomValue.cpp

```diff
@@ -0,0 +1,1305 @@
+    {
+    }
+
+    Value::Value(const Value& value)
+        : m_value(value.m_value)
+    {
+    }
+
+    Value::Value(Value&& value) noexcept
+    {
+        memcpy(this, &value, sizeof(Value));
+        memset(&value, 0, sizeof(Value));
+    }
+
+    Value::Value(AZStd::string_view stringView, bool copy)
+    {
+        if (copy)
+        {
+            CopyFromString(stringView);
+        }
+        else
+        {
+            SetString(stringView);
+        }
+    }
+
+    Value::Value(const AZStd::any& value)
+        : m_value(AZStd::allocate_shared<AZStd::any>(StdValueAllocator(), value))
+    {
+    }
+
+    Value Value::FromOpaqueValue(const AZStd::any& value)
+    {
+        return Value(value);
+    }
+
+    Value::Value(int8_t value)
+        : m_value(aznumeric_cast<int64_t>(value))
+    {
+    }
+
+    Value::Value(uint8_t value)
+        : m_value(aznumeric_cast<uint64_t>(value))
+    {
+    }
+
+    Value::Value(int16_t value)
+        : m_value(aznumeric_cast<int64_t>(value))
+    {
+    }
+
+    Value::Value(uint16_t value)
+        : m_value(aznumeric_cast<uint64_t>(value))
+    {
+    }
+
+    Value::Value(int32_t value)
+        : m_value(aznumeric_cast<int64_t>(value))
+    {
+    }
+
+    Value::Value(uint32_t value)
+        : m_value(aznumeric_cast<uint64_t>(value))
+    {
+    }
+
+    Value::Value(int64_t value)
+        : m_value(value)
+    {
+    }
+
+    Value::Value(uint64_t value)
+        : m_value(value)
+    {
+    }
+
+    Value::Value(float value)
+        : m_value(aznumeric_cast<double>(value))
+    {
+    }
+
+    Value::Value(double value)
+        : m_value(value)
+    {
+    }
+
+    Value::Value(bool value)
+        : m_value(value)
+    {
+    }
+
+    Value::Value(Type type)
+    {
+        switch (type)
+        {
+        case Type::Null:
+            // Null is the default initialized value
+            break;
+        case Type::Bool:
+            m_value = false;
+            break;
+        case Type::Object:
+            SetObject();
+            break;
+        case Type::Array:
+            SetArray();
+            break;
+        case Type::String:
+            SetString("");
+            break;
+        case Type::Int64:
+            m_value = int64_t{};
+            break;
+        case Type::Uint64:
+            m_value = uint64_t{};
+            break;
+        case Type::Double:
+            m_value = double{};
+            break;
+        case Type::Node:
+            SetNode("");
+            break;
+        case Type::Opaque:
+            AZ_Assert(false, "AZ::Dom::Value may not be constructed with an empty opaque type");
+            break;
+        }
+    }
+
+    Value& Value::operator=(const Value& other)
+    {
+        m_value = other.m_value;
+        return *this;
+    }
+
+    Value& Value::operator=(Value&& other) noexcept
+    {
+        SetNull();
+        memcpy(this, &other, sizeof(Value));
+        memset(&other, 0, sizeof(Value));
+        return *this;
+    }
```

Comment:
The stored `any`s are heap allocated, so the trivial move should work fine (since we're just copying a pointer and a ref counter). It's a fairly significant (~10MB/s deserialization throughput) perf gain, so I'm inclined to leave this but I'll keep an eye out for any issues here and make sure to unit/regression test anything that pops up.